controllers/numaresourcesscheduler_controller.go

```diff
@@ -152,11 +152,16 @@ func isDeploymentRunning(ctx context.Context, c client.Client, key nrsv1alpha1.N
 }
 
 func (r *NUMAResourcesSchedulerReconciler) syncNUMASchedulerResources(ctx context.Context, instance *nrsv1alpha1.NUMAResourcesScheduler) (nrsv1alpha1.NamespacedName, string, error) {
+	var deploymentNName nrsv1alpha1.NamespacedName
+	schedulerName := instance.Spec.SchedulerName
+
 	schedstate.UpdateDeploymentImageSettings(r.SchedulerManifests.Deployment, instance.Spec.SchedulerImage)
 	schedstate.UpdateDeploymentConfigMapSettings(r.SchedulerManifests.Deployment, r.SchedulerManifests.ConfigMap.Name)
+	err := schedstate.UpdateSchedulerName(r.SchedulerManifests.ConfigMap, instance.Spec.SchedulerName)
```

Comment:
let's make this more explicit: if not name was set excplicitely, don't update and stick with whatever default we have in the YAML asset. So let's move the check `if schedulerName != '"` here in the calling site...